server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -2010,6 +2010,7 @@ public void forceMerge(
         } catch (Exception e) {
             try {
                 maybeFailEngine("force merge", e);
+                indexWriter.flush();
```

Comment:
We are invoking a second merge within the context of a failed merge, which might be tricky and non-deterministic. Essentially we are  saying that a second merge(merge + delete new files) will succeed when the first one failed.

Reply:
Had raised the same - https://github.com/opensearch-project/OpenSearch/pull/6324#discussion_r1106684057. @RS146BIJAY  you had mentioned that the merge does not get triggered when in exception block. Hoping that we have confirmed the same.

Reply:
As of now, lucene only allows a [flush call](https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/index/IndexWriter.java#L4145) where a regular segment merge is triggered after we delete new lucene files. It does not exposes another overload of [flush(boolean triggerMerge, boolean applyAllDeletes)](https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/index/IndexWriter.java#L4156) (where we can set triggerMerge to false to not allow segment merge along with file deletion).

As per the [corresponding Lucene issue link](https://issues.apache.org/jira/browse/LUCENE-6849), this is kept intentionally that if we delete some file we perform a regular segment merge after it.

> +1 on the patch and agreed that we should only expose a no-argument flush method, it's actually nice that the patch decreased the visibility of flush(boolean, boolean).

> I don't think we should add anything but a no-arg flush() here.
If its going to be more engineered than that, then this is the wrong direction: lets just not expose it at all.